production/helm/promtail/templates/daemonset.yaml

```diff
@@ -37,7 +37,11 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
             - "-config.file=/etc/promtail/promtail.yaml"
-            - "-client.url=http://{{ include "loki.serviceName" . }}:{{ .Values.loki.servicePort }}/api/prom/push"
+            {{- if .Values.loki.useBasicAuth }}
```

Comment:
Instead of an extra boolean, can probably just use `{{- if and .Values.loki.user .Values.loki.password }}`. That way it saves the extra variable and also checks if user and password is non-empty.